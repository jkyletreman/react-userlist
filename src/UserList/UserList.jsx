import React from "react";
import UserData from "../data/context";

export const UserList = () => {
  return (
    <UserData.Consumer>
      {context => {
        return context.state.userData.map(user => {
          return (
            <li key={user.id} onClick={context.selectUser}>
              {user.first} {user.last}
            </li>
          );
        });
      }}
    </UserData.Consumer>
  );
};

export default UserList;
