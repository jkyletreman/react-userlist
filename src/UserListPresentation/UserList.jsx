import React from "react";
import UserData from "../context";

const UserList = ({ userData, selectUser}) => {
  return (
    <ul>
      {userData.map(user => {
        return (
          <li key={user.id} onClick={selectUser}>
            {user.first} {user.last}
          </li>
        );
      })}
    </ul>
  );
};

export default UserList;
