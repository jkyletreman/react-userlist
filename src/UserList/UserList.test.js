import React from "react";
import ReactDOM from "react-dom";
import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import UserList from "./UserList";
const userData = require("../data/userData.json");

configure({ adapter: new Adapter() })


describe("<UserList />", () => {
  const userlist = shallow(<UserList />);

  it("renders shallow correctly", () => {
    console.log(userlist.debug())
    expect(toJson(userlist)).toMatchSnapshot();
  });

  it('should accept `userData as props`', () => {
    expect(userlist.props().userData).toEqual(userData)
  });

  it("should render a ul with 3 li children", () => {
    expect(userlist.find('ul').children().length).toBe(3);
  });
});
