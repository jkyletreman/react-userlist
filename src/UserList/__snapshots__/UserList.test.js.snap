// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> renders shallow correctly 1`] = `
<UserList
  userData={
    Array [
      Object {
        "age": 51,
        "description": "John is a retired YouTuber",
        "first": "John",
        "id": 1,
        "last": "Roberts",
        "location": "Chicago, IL",
      },
      Object {
        "age": 27,
        "description": "Frita loves the desert, cheese, and turtles.",
        "first": "Frita",
        "id": 2,
        "last": "Waters",
        "location": "Phoenix, AZ",
      },
      Object {
        "age": 24,
        "description": "Bart skateboards everywhere.",
        "first": "Bart",
        "id": 3,
        "last": "Simpson",
        "location": "Springfield, IL",
      },
    ]
  }
>
  <ul>
    <li
      key="1"
    >
      John
       
      Roberts
    </li>
    <li
      key="2"
    >
      Frita
       
      Waters
    </li>
    <li
      key="3"
    >
      Bart
       
      Simpson
    </li>
  </ul>
</UserList>
`;
